pkg/skaffold/docker/parse.go

```diff
@@ -77,10 +76,12 @@ func readCopyCmdsFromDockerfile(onlyLastImage bool, absDockerfilePath, workspace
 		return nil, fmt.Errorf("parsing dockerfile %q: %w", absDockerfilePath, err)
 	}
 
+	// TODO: re-enable this after updating moby/buildkit dependency
+	// https://github.com/GoogleContainerTools/skaffold/issues/5049
 	// instructions.Parse will check for malformed Dockerfile
-	if _, _, err := instructions.Parse(res.AST); err != nil {
-		return nil, fmt.Errorf("parsing dockerfile %q: %w", absDockerfilePath, err)
-	}
+	// if _, _, err := instructions.Parse(res.AST); err != nil {
```

Comment:
@shaxbee Unfortunately, skaffold needs to parse the dockerfile to get 
1. dependencies to watch
2. to create a docker context etc.

It would make sense to update the docker dependency.

Reply:
@tejal29 I think there are actually two call to different `Parse` functions. The one that is modified here is the one that just checks for malformed Dockerfile commands. The one that actually gets the required information from the dockerfile is still intact

Reply:
Exactly.

Reply:
Actually instructions are used to resolve ONBUILD so this cannot be disabled.

Reply:
resolving this as it's no longer relevant to the PR